# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/base/util-linux/nice-mount-paths.patch
# ROCK Linux is Copyright (C) 1998 - 2006 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- ./mount/mount.c.orig	2004-12-30 17:59:42.242468848 +0100
+++ ./mount/mount.c	2004-12-30 18:00:26.571729776 +0100
@@ -1159,6 +1159,23 @@
 	/* Substitute values in opts, if required */
 	opts = usersubst(opts);
 
+	/* make the node a nice absolute path name */
+	if (node[0] && node[0]!='/') {
+	  char *thisdir=get_current_dir_name();
+	  char *oldnode=node;
+	  if (chdir(node) == 0) {
+	    /* we will never free this variable ... */
+	    node=get_current_dir_name();
+	    chdir(thisdir);
+	    if (verbose) printf("mount: change '%s' to '%s'\n",oldnode,node);
+	  }
+	  free(thisdir);
+	}
+	if (strlen(node)>=3 && node[strlen(node)-1]=='/') {
+	  if (verbose) printf("mount: remove the leading '/' in '%s'\n",node);
+	  node[strlen(node)-1]=0;
+	}
+
 	/* Merge the fstab and command line options.  */
 	if (opts == NULL)
 		opts = cmdlineopts;
